### DISCUSSION TITLE
Add `@starting-style` variant

### DISCUSSION THREAD

**Discussion by nix6839** [tailwindlabs/tailwindcss#12039] [Ideas]
This allow authors to start CSS transitions on first style update.

More info about this: https://developer.chrome.com/en/blog/new-in-chrome-117/#exit-entry-animations

**Comment by JeroenReumkens** [comment]
Just noticed this has been added to v4 alpha. So perhaps they won't be adding it to v3 anymore, but v4 will have it! 

https://github.com/tailwindlabs/tailwindcss/pull/13329

**Reply by twilkinson3421** [reply]
If you want to use the variant now, you can add the following to the plugins array in your _tailwind_ config:

```ts
/** @type {import('tailwindcss/types/config').PluginCreator} */
({ addVariant }: any) => {
  addVariant("starting", "@starting-style");
},
```

Then you can use the `starting:` variant in your _tailwind_ styles.

**Reply by JeroenReumkens** [reply]
YES! I made a video about it last week, after figuring out how it works. Only forgot to update it here too 😂 
Thanks for adding the comment here as well! 

https://youtu.be/jCqtngrL2pA?si=Co0Ajzs0uPo5ju3T

**Reply by 6TELOIV** [reply]
For better typing:
```ts
import type { PluginAPI } from "tailwindcss/types/config";
/* ... */
({ addVariant }: PluginAPI) => {
  addVariant("starting", "@starting-style");
}
```
